### PR TITLE
Rework scene creation dialog

### LINKED ISSUE #34524: FileSystem -> Create Scene asks for a name, but it is not used
**Godot version:**
3.2 Beta 4

**OS/device including version:**
Mint 19.2
Linux Linux 4.15.0-72-generic

**Issue description:**
image
image

**Steps to reproduce:**
Right click in FileSystem to create a new scene and it asks for a name. This should just be removed completely because it makes an empty scene anyways.

### PR DESCRIPTION
**Pull request by KoBeWi**
Replaces this thing 
image

With this 😎
godot windows tools 64_GMh3E1bbRs

Closes #34524

### CODE REVIEW

File: editor/scene_create_dialog.cpp [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+		scene_name->connect("text_changed", callable_mp(this, &SceneCreateDialog::update_dialog).unbind(1));
+	}
+
+	{
+		Label *label = memnew(Label(TTR("Root Name:")));
+		gc->add_child(label);
+
+		root_name = memnew(LineEdit);
+		gc->add_child(root_name);
+		root_name->set_placeholder(TTR("Leave empty to use scene name."));
```
Comment (fire-forge) [inline editor/scene_create_dialog.cpp:216]:
```suggestion
		root_name->set_placeholder(TTR("Leave empty to use scene name"));
```

**Review by fire-forge — COMMENTED** [review]
I think it could use a bit of padding here, maybe 4px:

image

Also instead of always using the regular Node icon for custom type, it could change to the icon of whatever node is selected.

Aside from that, the new design is pretty good!

**Comment by Ansraer** [comment]
Hmm, the indentation of the radio buttons feels wrong to me when compared to `scene name` and `root name` fields. Maybe some padding between the last radio option and `scene name` could help?
Or maybe try moving the radio buttons below the `root name ` field. Having the top fields look less indented than the bottom ones will probably look better.

Also, I think you should remove the dot in the `root name` placeholder text. Most UIs I can think of don't use punctuation in input hints.

File: editor/scene_create_dialog.cpp [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+		hb->add_child(node_type_custom);
+		node_type_custom->set_button_group(node_type_group);
+
+		Control *spacing = memnew(Control);
+		hb->add_child(spacing);
+		spacing->set_custom_minimum_size(Size2(4 * EDSCALE, 0));
+
+		root_custom_type = memnew(LineEdit);
+		hb->add_child(root_custom_type);
+		root_custom_type->set_placeholder(TTR("Custom Node"));
```
Comment (aaronfranke) [inline editor/scene_create_dialog.cpp:207]:
```suggestion
		root_other_type->set_placeholder(TTR("Other Node"));
```
"Custom" here might give the impression that it only allows user-created node types.

Comment (KoBeWi) [reply editor/scene_create_dialog.cpp:207]: Funny that the root option in empty scene was also called "Custom Node" until I changed it in #26970

**Comment by KoBeWi** [comment]
Ok seems finished:
godot windows tools 64_tOfS4p3auQ
Doing code cleanup rn.

**Comment by fire-forge** [comment]
What do you think about using a dropdown list/OptionButton for the scene file extension? I noticed it only allows `.tscn` and `.scn` and not any arbitrary extensions, so that would make it easier to know what extensions are allowed and would remove the need for the invalid extension warning message.

**Comment by Ansraer** [comment]
I really think this would look better if you moved scene name and root name above root type.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by Sslaxx** [comment]
Any chance of this being cherrypicked?

**Comment by fire-forge** [comment]
> IMO it would be better if the "root type" showed the actual node names. Node2D / Control etc. would be more immediately recognizable to any non-beginner.

I personally think the buttons are fine the way they are now. It describes the intent of the scene (2D, 3D, or UI) instead of the actual node type used.

> Also the cancel/confirm buttons should be swapped.

The cancel and ok buttons order is different depending on the platform. So whichever platform you're on must use a different order than the platform KoBeWi took the screenshots on. See this quote from https://github.com/godotengine/godot/pull/59382:

> The position (left/right) of the Cancel and OK buttons in AcceptDialog
are DisplayServer specific, as Windows uses OK/Cancel and macOS uses Cancel/OK.

**Pull request merged**